### PR TITLE
Refactor/country

### PR DESCRIPTION
**Pull request by necoline**
- [x] updates components to hooks
- [x] makes requests for country and locations in the component and removes redux actions
- [x] adds tests
- [x] adds header stats to common component
- [x] adds a few other common components
- [ ] moves the map to separate component - started but not complete

### CODE REVIEW

File: app/assets/scripts/views/country/country.js [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    };
+
+    const fetchCountry = id => {
+      setCountry(state => ({
+        ...state,
+        countryFetching: true,
+        countryError: null,
+      }));
+      let limit = 1;
+      fetch(`${config.api}/countries?limit=${limit}&country=${id}`)
```
Comment (AliceR) [inline app/assets/scripts/views/country/country.js:84]: Maybe we should change the /countries endpoint to accept `​/countries​/{country_id}`, for consistency with /locations and /projects. What do you think, @bitner ?

**Comment by AliceR** [comment]
Almost done, this looks very good already. Don't worry too much about the map component, it should be tackled in #476 .
Just the lint and a little bit of cleanup (old proptypes, unused selector) and this is good to merge.

File: app/assets/scripts/views/country/country.js [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+      ).map(JSON.parse)
+    : null;
+
+  let locationGroups = _(locations).sortBy('city').groupBy('city').value();
+
+  return (
+    <>
+      {(countryError || !country) && <ErrorHeader />}
+      {countryFetching && <LoadingHeader />}
+      {country && locations && (
```
Comment (AliceR) [inline app/assets/scripts/views/country/country.js:139]: should we use some if/else ?/: to ensure error, loading, and actual countries page are mutually exclusive?

Comment (necoline) [reply app/assets/scripts/views/country/country.js:139]: ah yes, that is better

**Pull request merged**